src/main/java/org/hisp/dhis/rules/functions/RuleFunction.java

```diff
@@ -9,115 +12,128 @@
+            return RuleFunctionZScoreHFA.create();
+        case RuleFunctionZScoreWFH.D2_ZSCOREWFH:
+            return RuleFunctionZScoreWFH.create();
+        case RuleFunctionMaxValue.D2_MAX_VALUE:
+            return RuleFunctionMaxValue.create();
+        case RuleFunctionMinValue.D2_MIN_VALUE:
+            return RuleFunctionMinValue.create();
+        case RuleFunctionZing.D2_ZING:
+            return RuleFunctionZing.create();
+        case RuleFunctionOizp.D2_OIZP:
+            return RuleFunctionOizp.create();
+        case RuleFunctionCountIfZeroPos.D2_COUNT_IF_ZERO_POS:
+            return RuleFunctionCountIfZeroPos.create();
+        case RuleFunctionLeft.D2_LEFT:
+            return RuleFunctionLeft.create();
+        case RuleFunctionRight.D2_RIGHT:
+            return RuleFunctionRight.create();
+        case RuleFunctionValidatePattern.D2_VALIDATE_PATTERN:
+            return RuleFunctionValidatePattern.create();
+        case RuleFunctionConcatenate.D2_CONCATENATE:
+            return RuleFunctionConcatenate.create();
+        case RuleFunctionInOrgUnitGroup.D2_IN_ORG_UNIT_GROUP:
+            return RuleFunctionInOrgUnitGroup.create();
+        case RuleFunctionHasUserRole.D2_HAS_USER_ROLE:
+            return RuleFunctionHasUserRole.create();
+        default:
+            return null;
         }
+    }
 
-        @Nonnull
-        public double toDouble( @Nullable final String str, final double defaultValue )
+    public TimeInterval getTimeInterval( String start, String end )
+    {
+        if ( isEmpty( start ) || isEmpty( end ) )
         {
-                if ( str == null )
-                {
-                        return defaultValue;
-                }
-
-                try
-                {
-                        return Double.parseDouble( str );
-                }
-                catch ( final NumberFormatException nfe )
-                {
-                        return defaultValue;
-                }
+            return TimeInterval.empty();
         }
 
-        @Nonnull
-        public String wrap( String input )
+        LocalDate startDate = LocalDate.parse( start, DateTimeFormat.forPattern( DATE_PATTERN ) );
+        LocalDate endDate = LocalDate.parse( end, DateTimeFormat.forPattern( DATE_PATTERN ) );
+
+        return TimeInterval.fromTo( startDate, endDate );
+    }
+
+    @Nonnull
+    public double toDouble( @Nullable final String str, final double defaultValue )
+    {
+        if ( str == null )
         {
-                if( input == null )
-                {
-                        return "";
-                }
+            return defaultValue;
+        }
 
-                return "'"+input+"'";
+        try
+        {
+            return Double.parseDouble( str );
         }
+        catch ( final NumberFormatException nfe )
+        {
+            return defaultValue;
+        }
+    }
 
-        static boolean isEmpty( String input )
+    @Nonnull
+    public String wrap( String input )
```

Comment:
Perhaps trim single quotes, or check that we do not double wrap - if a user calls the function with hard coded dates with single quotes:
d2:daysBetween(#{event_Date},'2020-01-01')

Reply:
I added a test to verify that single quoted dates are parsed in the correct way.
This wrap is used for output, and it is actually one of the changes that we need to verify in the Antlr PR